src/request/request.service.ts

```diff
@@ -0,0 +1,132 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/sequelize';
+import { CreateRequsetDto } from './dto/create-request.dto';
+import { Request } from './request.model';
+import { mailer } from '../nodemailer';
+import { UserService } from 'src/user/user.service';
+import { RECIPIENT_NOT_FOUND, RESENDING } from 'src/constants';
+import { TeamService } from 'src/team/team.service';
+import { User } from 'src/user/user.model';
+
+@Injectable()
+export class RequestService {
+  constructor(@InjectModel(Request) private requestRepository: typeof Request, 
+  private userService: UserService,
+  private teamService: TeamService) {}
+
+  async getMyNotifications(req: any) {
+    return await this.requestRepository.findAll( { where: {to: req.user.id}})
+  }  
+
+  async requestJoinTeam(req: any, input: CreateRequsetDto) {
+    // or recievd only 'team' in input and search manager throught team ?????????????????
+    let manager;
+    try {
+      manager = await this.userService.getUserById(input.to);
+    } catch(e) {
+      throw new HttpException(RECIPIENT_NOT_FOUND, HttpStatus.NOT_FOUND);
+    }
+
+    const requestMessage = {
+      from: req.user.id,
+      to: input.to,
+      type: 'join',
+      description: `team: ${input.team} , player: ${req.user.email}`,
+    }
+    
+    const message = {
+      to: manager.email,
+      subject: 'confirm join',
+      text: '',
+      html: `
+      <p>confirm joining team ${input.team} of player ${req.user.email}</p>
+  `,
+    };
+
+    const request = await this.requestRepository.findAll({ where: {
+      description: requestMessage.description, status: 'decline',type: requestMessage.type}
+    });
+
+    if (request.length === 0) {
+      mailer(message);
```

Comment:
if there are no requests, send an email and write into db. It seems weird to me

Reply:
This is a check to resend the request. If it doesn't exist yet, the request is created and an email is sent. Otherwise, an error that the request has already been sent